src/captiveportal/captiveportalrequest.cpp

```diff
@@ -57,6 +57,13 @@ void CaptivePortalRequest::createRequest(const QUrl& url) {
   NetworkRequest* request = NetworkRequest::createForCaptivePortalDetection(
       this, url, CAPTIVEPORTAL_HOST);
 
+  
+  connect(request,&NetworkRequest::requestRedirected,[this](NetworkRequest* request, const QUrl& url){
+    // In Case the Captive Portal request Redirects, we 100% have one. 
+    logger.info() << "Portal Detected -> Redirect to " <<url.toString();
+    request->abort();
+    onResult(PortalDetected);
+  });
```

Comment:
This popped up while testing, when we have a redirect to something that requires dns it is not available (as we might use the tunneled one) - the request will fail with "hostnotfound". The intent to redirect is all we need :)